src/prefect/environments/storage/docker.py

```diff
@@ -165,7 +166,7 @@ def name(self) -> str:
             raise ValueError("Docker storage is missing required fields")
 
         return "{}:{}".format(
-            os.path.join(self.registry_url, self.image_name),  # type: ignore
+            PurePosixPath(self.registry_url, self.image_name),  # type: ignore
```

Comment:
Why is the other `str` and not this one?

Reply:
This one implicitly gets converted to a string through the `.format()`, the other one I had to cast myself.